x/incentive/types/rewards.go

```diff
@@ -72,9 +104,51 @@ func NewClaimPeriod(denom string, id uint64, end time.Time, timeLock time.Durati
 	}
 }
 
+// Validate performs a basic check of a ClaimPeriod fields.
+func (cp ClaimPeriod) Validate() error {
+	if cp.ID == 0 {
+		return errors.New("claim period id cannot be 0")
+	}
+	if cp.End.IsZero() {
+		return errors.New("claim period end time cannot be 0")
+	}
+	if cp.TimeLock == 0 {
+		return errors.New("claim period time lock cannot be 0")
+	}
+	return sdk.ValidateDenom(cp.Denom)
+}
+
+// String implements fmt.Stringer
+func (cp ClaimPeriod) String() string {
+	return fmt.Sprintf(`Claim Period:
```

Comment:
You can (should?) just use YAML here to make this cleaner.

Reply:
Agree, there's an issue for that #492

Reply:
But it's a code change you're adding here. It'd be one thing of the code already existed.

Reply:
I just moved it. But I can update it to use yaml 👍